components/TextFields/src/MDCTextInputBorderView.m

```diff
@@ -0,0 +1,171 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCTextInputBorderView.h"
+
+static NSString *const MDCTextInputBorderViewBorderFillColorKey =
+    @"MDCTextInputBorderViewBorderFillColorKey";
+static NSString *const MDCTextInputBorderViewBorderPathKey =
+    @"MDCTextInputBorderViewBorderPathKey";
+static NSString *const MDCTextInputBorderViewBorderStrokeColorKey =
+    @"MDCTextInputBorderViewBorderStrokeColorKey";
+
+static inline NSString * _Nullable MDCNSStringFromCGLineCap(CGLineCap lineCap) {
+  NSString *lineCapString;
+  switch (lineCap) {
+    case kCGLineCapButt:
+      lineCapString = kCALineCapButt;
+      break;
+    case kCGLineCapRound:
+      lineCapString = kCALineCapRound;
+      break;
+    case kCGLineCapSquare:
+      lineCapString = kCALineCapSquare;
+      break;
+    default:
+      break;
+  }
+  return lineCapString;
+}
+
+static inline  NSString *_Nullable MDCNSStringFromCGLineJoin(CGLineJoin lineJoin) {
+  NSString *lineJoinString;
+  switch (lineJoin) {
+    case kCGLineJoinBevel:
+      lineJoinString = kCALineJoinBevel;
+      break;
+    case kCGLineJoinMiter:
+      lineJoinString = kCALineJoinMiter;
+      break;
+    case kCGLineJoinRound:
+      lineJoinString = kCALineJoinRound;
+      break;
+    default:
+      break;
+  }
+  return lineJoinString;
+}
+
+@interface MDCTextInputBorderView()
+
+@property(nonatomic, strong, readonly) CAShapeLayer *borderLayer;
+
+@end
+
+@implementation MDCTextInputBorderView
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCTextInputBorderViewInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder
+{
+  self = [super initWithCoder:coder];
+  if (self) {
+    _borderFillColor = [coder decodeObjectForKey:MDCTextInputBorderViewBorderFillColorKey];
+    _borderPath = [coder decodeObjectForKey:MDCTextInputBorderViewBorderPathKey];
+    _borderStrokeColor = [coder decodeObjectForKey:MDCTextInputBorderViewBorderStrokeColorKey];
+    [self commonMDCTextInputBorderViewInit];
```

Comment:
commonMDCXYZInit before decoded values or decoded values will get overwritten by defaults set in commonInit.
Looking at commonInit, this will work as intended.
We should determine the 'one, true way' we want to initWithCoder.
Happy to do so in another Issue / meeting / PR.